### PR TITLE
Add support for customising the DayCell

### PR DESCRIPTION
**Pull request by kamyar**
What types of changes does your code introduce?

- [x] New feature (non-breaking change which adds functionality)

## Description
Adds support for passing a function to customize DayCell content.

> Related Issue: #242

### CODE REVIEW

**Comment by brunolucena** [comment]
+1

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,14562 @@
+{
```
Comment (mkg0) [inline package-lock.json:1]: I guess it's not intentional, this project uses yarn.
If not, we should consider this kinda change in a different PR with other related updates. I would prefer to have a test setup before applying this change tho :)

**Review by mkg0 — CHANGES_REQUESTED** [review]
LGTM except for one thing

File: package-lock.json (hunk 1 above)
Comment (kamyar) [reply package-lock.json:1]: Ah, yes, have missed this file, will remove. 👍

**Review by onionhammer — APPROVED** [review]
Looks great!

**Review by mkg0 — APPROVED** [review]

**Review by charithAmila — COMMENTED** [review]
Looks good

**Comment by okkindel** [comment]
+1

**Comment by HugoLiconV** [comment]
Hi everyone! Wouldn't be useful to implement something similar to this but for the whole button, not just for the content? So instead of doing this: 
```javascript
<button
// ...buttonProps
>
  {dayContentRenderer?.(this.props.day) || (
    <span>{format(this.props.day, this.props.dayDisplayFormat)}</span>
  )}
</button>
```
We'll be doing this: 
```javascript
if (dayContentRenderer) {
  return dayContentRenderer({  
    ...buttonProps
  });
}
return (
  <button
     // ...buttonProps
  >
    <span>{format(this.props.day, this.props.dayDisplayFormat)}</span>
  </button>
)
```
What do you think?

**Pull request merged**